Secway and janicart key suicides

ArcaneDefence opened the pull request:
## About The Pull Request

Adds a unique suicide to the secway key, where the user spins and then gibs.
Adds a unique suicide to the janitor cart key, where it checks your skill level and does a thematic suicide.
If you have no cleaning level, or a novice level, you immediately gib.
If you are an apprentice or journeyman you clean the tiles you walk on and then die in 5 seconds.
If you are an expert or a master you flash janicart colors and clean the tiles you walk on and then die in 10 seconds.
If you are a legendary janny though, you have a windup noise, flash janicart colors, clean tiles you walk on, gain 10u of meth in your system, and die in 15 seconds.
As an added bonus legendary jannies bwoink when their time as a janicart expires.

## Why It's Good For The Game

I was extremely disappointed to find that there was no suicide for this unique item, and decided to come up with my own idea.

Being asked why I didn't do the janicart with the secway made me think of something better, and more thematic.
I, for one, respect the janitorial arts.

## Changelog
:cl:
add: The secway key now has a unique suicide
add: The janicart key now has a uniquer suicide that respects janitorial skill
/:cl:

ArcaneDefence committed:
Update vehicle_key.dm

Adds a unique suicide to the secway key where the user spins, and gibs after a short delay.

ATH1909 commented on code/modules/vehicles/vehicle_key.dm:
```diff
@@ -9,6 +9,12 @@
 
+/obj/item/key/security/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] is putting \the [src] in [user.p_their()] ear and begins spinning! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	user.emote("spin")
```
add a shame suicide if they're somehow unable to spin

ArcaneDefence replied:
Can't say I know how to do that

ArcaneDefence replied:
Got that figured out.

ArcaneDefence committed:
Improvements

Fixes a runtime, adds a little silliness.

ArcaneDefence committed:
Adds shame result

Grammar changes and "rewards" anyone that successfully fails to commit suicide.

Dennok commented:
Nice but why not for janitorial key too?

PKPenguin321 commented:
This is pretty close to the hand drill suicide I added but with less spin

ArcaneDefence commented:
I wanted to put something silly to the secway key because the secway inspires images of hopcurity, RDsec, and dufflesec. I don't really think I want to mock the jannycart key with this suicide, as using the floor buffer is genuinely good.

ArcaneDefence committed:
Adds janicart key suicide

Adds a skillcheck to the suicide that will have varying effects based on the level the user has.

ArcaneDefence committed:
Adds the return at the end of the file back

Whoops, didn't notice I deleted that I guess. Dunno if that matters, but I have no reason to remove it, so I'm putting it back.

Pumpkinoe commented:
does curator's case key have a unique suicide

ArcaneDefence commented:
> 
> 
> does curator's case key have a unique suicide

No, it does not. The only thing I can think of is to have it create a display case with your head in it, but those cases set off alarms when smashed. Not really something I want to touch right now.

ATH1909 commented on code/modules/vehicles/vehicle_key.dm:
```diff
@@ -9,10 +9,57 @@
+/obj/item/key/proc/manual_suicide(mob/living/user)
+	if(user)
+		user.remove_atom_colour(ADMIN_COLOUR_PRIORITY)
+		user.visible_message("<span class='suicide'>[user] forgot [user.p_they()] isn't actually a janicart! That's a paddlin'!</span>")
```
"they isn't"?

ArcaneDefence replied:
If I'm told they is a singular pronoun I'm going to use it as a singular pronoun.
He isn't, she isn't, they isn't.
/shrug

MrDoomBringer commented:
based jani suicide

Dennok commented:
> > does curator's case key have a unique suicide
> 
> No, it does not. The only thing I can think of is to have it create a display case with your head in it, but those cases set off alarms when smashed. Not really something I want to touch right now.

Case can be created alarmless, see case construction, it must create alarmless case, if it PR merged.
